### PR TITLE
Todo addition and completion feature with a menu for todo status

### PR DESCRIPTION
**Pull request by nick-chal**
Addition and basic toggle added.
Three options shown
Remaining - delete, edit and search

### CODE REVIEW

File: src/assets/css/styles.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+import './style.css';
+import './App.css';
+import './reset.css';
```
Comment (SafalPandey) [inline src/assets/css/styles.js]: This import doesn't look right. You should import `reset.css` before importing other css files.

File: src/components/App.js [hunk 2]
```diff
@@ -1,9 +1,16 @@
 import React, { Component } from 'react';
 import '../assets/css/App.css';
```
Comment (SafalPandey) [inline src/components/App.js]: You're importing `App.css` twice here. Once in this line and then again in `styles.js`.

File: src/components/TodoForm.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        text: this.state.text,
+        completed: false
+      });
+      this.setState({ text: '' });
+    }
+  };
+
+  render() {
+    return (
+      <>
```
Comment (SafalPandey) [inline src/components/TodoForm.js]: No need to use fragments here.

File: src/components/TodoForm.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  updateOnChange = e => {
+    const { value, name } = e.target;
+    this.setState({
+      [name]: value
+    });
+  };
+
+  submitTodo = e => {
+    e.preventDefault();
+    if (/\S/.test(this.state.text)) {
```
Comment (SafalPandey) [inline src/components/TodoForm.js]: Add a newline before if statements.
Create a utility function like `isValidText` that wraps this logic to make it more readable.

File: src/components/TodoList.js [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+  populateTodo = showState => {
+    if (showState === 'remaining')
+      return this.state.todos.filter(todo => !todo.completed);
+    else if (showState === 'completed')
+      return this.state.todos.filter(todo => todo.completed);
+    else return this.state.todos;
+  };
+
+  toggleShowState = showState => {
```
Comment (SafalPandey) [inline src/components/TodoList.js]: `set` would be a better name than `toggle`.

File: src/components/TodoList.js [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+    this.setState({ showState });
+  };
+
+  render() {
+    const renderTodo = this.populateTodo(this.state.showState);
+    return (
```
Comment (SafalPandey) [inline src/components/TodoList.js]: Add newline before return statement.

File: src/components/TodoList.js [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+      <>
+        <div>
+          <button
```
Comment (SafalPandey) [inline src/components/TodoList.js]: If you're keeping all the buttons, form and the list under this component, the name `TodoList` becomes a bit misleading.
Try moving everything except the actual list to `App` component you'll learn a thing or two in the process. :slightly_smiling_face:

**Review by SafalPandey — CHANGES_REQUESTED** [review]

File: src/components/TodoList.js [hunk 8]
```diff
@@ -0,0 +1,81 @@
+import React from 'react';
```
Comment (cham11ng) [inline src/components/TodoList.js]: Add one line to separate package import and app import.

**Pull request closed without merging**